typingsgen.js

```diff
@@ -0,0 +1,74 @@
+var fs = require('fs');
+var regex = /export interface .*?Operators<T> \{([\S|\s]*)\}/;
+
+var core = fs.readFileSync('./src/CoreOperators.ts').toString();
+var kitchenSink = fs.readFileSync('./src/Rx.KitchenSink.ts').toString();
+var combinedMethods = core.match(regex)[1].trim() + '\n' + kitchenSink.match(regex)[1].trim();
+var contents = combinedMethods.split('\n');
+
+var operators = {};
+var fileResult = '/* tslint:disable:class-name */ /* tslint:disable:no-unused-variable */ /* tslint:disable:max-line-length */\n\
+import {Observable} from \'./Observable\';\n\
+import {ConnectableObservable} from \'./observable/ConnectableObservable\';\n\
+import {Scheduler} from \'./Scheduler\';\n\
+import {Notification} from \'./Notification\';\n\
+import {Subject} from \'./Subject\';\n\
+import {Observer} from \'./Observer\';\n\
+import {GroupedObservable} from \'./operator/groupBy-support\';\n\
+import {GroupByObservable} from \'./operator/groupBy\';\n\
+import {TimeInterval} from \'./operator/extended/timeInterval\';\n\
+import {ObservableOrPromise, ArrayOrIterator, _Selector, _IndexSelector, _SwitchMapResultSelector, _ObservableMergeMapProjector, _IteratorMergeMapProjector, _Predicate, _PredicateObservable, _Comparer, _Accumulator, _MergeAccumulator} from \'./types\';\n\n\
```

Comment:
It would still have to be maintained in some way most likely.  That said we have a few things we could do to change how it works.

Change it around such that we edit the existing file, instead of writing a new one.  That way if any new definitions bring in additional Observables / Operators that need to be referenced, we can just update the file itself, and no rely only on the file generation.